### PR TITLE
SCRUM-2248 update - AlleleGeneAssociation loading - bulk load deprecation & link to gene

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/model/entities/Allele.java [hunk 1]
```diff
@@ -58,7 +58,6 @@
 @ToString(exclude = { "alleleGeneAssociations", "alleleDiseaseAnnotations", "alleleMutationTypes", "alleleSymbol", "alleleFullName", "alleleSynonyms", "alleleSecondaryIds", "alleleInheritanceModes", "alleleFunctionalImpacts", "alleleGermlineTransmissionStatus", "alleleDatabaseStatus", "alleleNomenclatureEvents" }, callSuper = true)
 @AGRCurationSchemaVersion(min = "1.7.3", max = LinkMLSchemaConstants.LATEST_RELEASE, dependencies = { GenomicEntity.class }, partial = true)
 @Table(indexes = { @Index(name = "allele_inCollection_index", columnList = "inCollection_id"), })
-@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "curie")
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/model/entities/Allele.java:61]: This annotation shouldn't have been kept in and is causing the error on the disease annotation table at the moment.

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java [hunk 2]
```diff
@@ -80,20 +91,40 @@ public List<Long> getAlleleGeneAssociationsByDataProvider(BackendBulkDataProvide
...
+				throw new ApiErrorException(response);
+			}
+			log.error(errorMessage);
+			return null;
+		}
+		if (deprecate) {
+			if (!association.getObsolete()) {
+				association.setObsolete(true);
+				if (authenticatedPerson.getOktaEmail() != null) {
+					association.setUpdatedBy(personService.findPersonByOktaEmail(authenticatedPerson.getOktaEmail()));
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java]: You should be able to use the authenticatedPerson as the object for setUpdatedBy... if not you should be able to use the authenticatedPerson.id for a personDao.get(id) as in authenticatedPerson is the object from the database. So it should be able to be used as such without another lookup.

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java [hunk 3]
```diff
@@ -80,20 +91,40 @@ public List<Long> getAlleleGeneAssociationsByDataProvider(BackendBulkDataProvide
...
+				} else {
+					association.setUpdatedBy(personService.fetchByUniqueIdOrCreate(loadDescription));
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java:113]: Are we creating a bunch of "persons" based on loadDescriptions? I don't think this is a good idea as in the person table is not really "persons" anymore?

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java:113]: Yeah, not sure what the best solution is without a model change though.  We've been doing the same for disease annotations as we want the "UpdatedBy" to indicate why the annotation was changed to obsolete and we don't have a person associated with a bulk load.

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java (hunk 2 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java]: Updated here and in the DiseaseAnnotationService

File: src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java (hunk 3 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java:113]: We also create "people" from whatever gets submitted in the updatedBy and createdBy fields in bulk submissions, which are often not really people either.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**